src/Shared/BuildEnvironmentHelper.cs

```diff
@@ -77,7 +77,12 @@ private static BuildEnvironment Initialize()
                 ()=> TryFromFolder(msbuildFromVisualStudioRoot, runningTests, runningInVisualStudio, visualStudioPath),
 
                 // Try from the current directory
-                () => TryFromFolder(currentDirectory, runningTests, runningInVisualStudio, visualStudioPath)
+                () => TryFromFolder(currentDirectory, runningTests, runningInVisualStudio, visualStudioPath),
+
+#if !CLR2COMPATIBILITY // Assemblies compiled against anything older than .NET 4.0 won't have a System.AppContext
+                // Try the base directory that the assembly resolver uses to probe for assemblies.
+                () => TryFromFolder(AppContext.BaseDirectory, runningTests, runningInVisualStudio, visualStudioPath)
```

Comment:
I went back and forth on this too.  Technically if I see MSBuild.exe (and the runtime) in the current directory, I sort of expect it to be used.  That would make sense for an end user but not necessarily for the app developer.  It seems rare that someone would change the current directory to some place that has MSBuild and then run a completely different app...